src/app/functionality/check_points_logic.py

```
@@ -0,0 +1,25 @@
+        try:
+            challenge_id = account['challengeId']
+            challenge_info = db_handler.getChallengeById(challenge_id)
+            challenge_name = challenge_info['name']
+
+            login(device, account['email'], account['password'])
+            new_points = check_points(device, challenge_name)
+            db_handler.updateAccount(account['id'], newPoints=new_points)
+            logout(device)
+        except ValueError as e:
+            # In the future the universal runner will allow to continue even if one account fails
+            raise ValueError(f"Failed to check points for account {account['username']}: {e}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)